### DISCUSSION TITLE
Error: "You're not authorized."

### DISCUSSION THREAD

**Discussion by chiruleonard** [sickmz/microw#13] [Q&A] [answered]
Hello -- I follow your installation steps and when I try to run I get "You're not authorized."

My .venv file looks like this:

```
SPREADSHEET_ID=1fRO94T-j2l_qyUbYP7nn1UkI28zXVJBZBk_**********
BOT_TOKEN=7110004342:AAGemGNv_**********
EXPENSE_SHEET=Expenses-spreadsheet
USER_ID=10692980280**********
```

Is there a problem with my .env file?

**Comment by sickmz** [comment, marked as answer]
> My .venv file looks like this:

Ehy, two points:
1. The secrets file should be called `.env`.
2. Try entering this log `print(update.effective_user.id, USER_ID)`  as the first instruction of the `start` function.

Something like this:

```
# Start command handler
async def start(update: Update, context: ContextTypes.DEFAULT_TYPE) -> int:
    print(update.effective_user.id, USER_ID)
    if str(update.effective_user.id) != str(USER_ID):
        await update.message.reply_text("You're not authorized. ⛔")
        return ConversationHandler.END
    await update.effective_message.reply_text("Hi! I'm microw. What can I do for you?", reply_markup=markup)

    return CHOOSING
```

Relaunch the script and check in the console that the two user ids are the same.

**Comment by chiruleonard** [comment]
Looks like there were different
```
693344***** 10692980280**********
```

Once I changed the .env file with the first one it worked. But I'm curious where this USER_ID is located in the Google Cloud Console. In credentials.json there's nothing.

**Comment by sickmz** [comment]
> Looks like there were different
> 
> ```
> 693344***** 10692980280**********
> ```
> 
> Once I changed the .env file with the first one it worked. But I'm curious where this USER_ID is located in the Google Cloud Console. In credentials.json there's nothing.

The `USER_ID` is your telegram user id. 
I will specify it better in the README, thank you.

**Reply by sickmz** [reply]
@chiruleonard closed in aabacb4fb4e666bc6eb228caf41d3b517b747126.
